### PR TITLE
Introduce pkg/ctrlz, Istio's introspection package.

### PR DESCRIPTION
**Pull request by geeknoid**
Processes that integrate with ControlZ open up a port that enables operators
to connect with a web browser and interact with the process. Through the browser,
the operator can adjust logging scope levels, see the process' command-line arguments
and envirinment variables, see statistics about heap use, and more.

Integration with ControlZ is nominally two line deal for processes. Optionally,
processes can extend the base ControlZ UI and integrate their own screens into the
main UI.

In addition to the browser interface, there is a REST API enabling access to all
the same things that the UI shows.

Mixer is integrated with ControlZ but doesn't currently have custom UI. We should
integrate ControlZ with our other server components in due time.

### CODE REVIEW

File: pkg/ctrlz/assets/templates/404.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    <div class="error">
+        We're sorry, the page you requested cannot be found
+    </div>
+
+    <div class="explanation">
+        The URL may be misspelled
+    </div>
+</div>
+{{ end }}
```
Comment (nmittler) [inline pkg/ctrlz/assets/templates/404.html]: nit: eol?

File: pkg/ctrlz/ctrlz.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+func registerTopic(router *mux.Router, layout *template.Template, t fw.Topic) {
+	htmlRouter := router.NewRoute().PathPrefix("/" + t.Prefix() + "z").Subrouter()
+	jsonRouter := router.NewRoute().PathPrefix("/" + t.Prefix() + "j").Subrouter()
+
+	tmpl := template.Must(template.Must(layout.Clone()).Parse("{{ define \"title\" }}" + t.Title() + "{{ end }}"))
+	t.Activate(fw.NewContext(htmlRouter, jsonRouter, tmpl))
+}
+
+// getLocalIP returns a non loopback local IP of the host
+func getLocalIP() string {
```
Comment (nmittler) [inline pkg/ctrlz/ctrlz.go]: Just a thought: I suspect this could be generally useful ... is there a common place we could put this?

**Comment by geeknoid** [comment]
@nmittler It'd be great to expose similar controls from the sidecars, but alas it would have to be completely separate code. I think the Envoy team is probably in a better position to deliver such a thing. Let's see how well this works for Istio components and maybe it'll be an obvious thing that other components will want to emulate.

**Comment by nmittler** [comment]
@geeknoid Yup .... sgtm, thanks!

**Pull request merged**